Include/Acidanthera/Library/OcHdaDevicesLib.h

```diff
@@ -58,6 +58,17 @@
 #define GET_CODEC_DEVICE_ID(a)  (a & 0xFFFFU)
 #define GET_CODEC_GENERIC_ID(a) (a | 0xFFFFU)
 
+//
+// Structure used for HDA PCI class code parsing.
+//
+#pragma pack(1)
+typedef struct {
+  UINT8 ProgInterface;
+  UINT8 SubClass;
+  UINT8 Class;
+} HDA_PCI_CLASSREG;
```

Comment:
I think this structure already exists in EDK II, and generally 4 bytes are read at once (including revision) through UINT32 read mode. Better to change.

Reply:
Is it the same thing as `USB_CLASSC` in EDK-II? If so, then it is indeed read as 3 bytes, but I can change to use that one instead.

Reply:
While it is, the one for USB semantically does not fit the use. I rather meant:

https://github.com/acidanthera/OpenCorePkg/blob/f61a1a35cbd12ec3758b0d510c0aec99d665abb2/Library/OcDeviceMiscLib/ActivateHpetSupport.c#L65
https://github.com/acidanthera/OpenCorePkg/blob/f61a1a35cbd12ec3758b0d510c0aec99d665abb2/Library/OcDeviceMiscLib/ResetAudioTrafficClass.c#L64
https://github.com/acidanthera/OpenCorePkg/blob/0.7.6/Library/OcDeviceMiscLib/SetResizableBar.c#L353

To be fair, maybe it is actually a good idea to change all four as the existing approach is not too great anyway. Could you define a PCI_CLASSCODE structure same as USB_CLASSC and switch to it everywhere?

Reply:
Hmm - just looking at the naming, I don't think `USB_CLASSC` can be the correct struct? (I think it must be the same thing, but named for a specific use-case?).

In which case, by hunting for references to the `PCI_CLASSCODE_OFFSET` in the EDK-2 code, we find that there is, indeed, no (other) struct for this in EDK-2, and also that it is always read as 3 bytes (not 4 bytes as one uint32).

So I think the net result is no change? Or possibly just change the naming `SubClass`->`SubClassCode` and `Class`->`ClassCode` which are used everywhere (both in `USB_CLASSC`, and also in various `UINT8` local variable names) in EDK-2.

Reply:
Sorry, we commented at the same time. Okay, will do.

Reply:
https://github.com/acidanthera/OpenCorePkg/pull/307/commits/4364ac86b926ce2ba5770969639bb15fcd2e3b71